src/App.jsx

```diff
@@ -1,20 +1,105 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import { useEffect, useState } from 'react'
+import { BrowserRouter, Route, Routes } from 'react-router-dom'
 import './App.css'
 import TodoList from './TodoList'
 import AddTodoForm from '../AddTodoForm'
 
 function App() {
-  const [newTodo, setNewTodo] = useState('')
+  let [todoList, setTodoList] = useState([])
+  let [isLoading, setIsLoading] = useState(true)
+
+  async function fetchData() {
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
+    }
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`
+
+    try {
+      const response = await fetch(url, options)
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`)
+      }
+
+      const data = await response.json()
+
+      const todos = data.records.map((record) => ({
+        id: record.id,
+        title: record.fields.title,
+      }))
+
+      setTodoList(todos)
+      setIsLoading(false)
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [])
+
+  useEffect(() => {
+    const loadTodoList = () => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const savedTodoList = localStorage.getItem('savedTodoList')
+          const initialList = savedTodoList ? JSON.parse(savedTodoList) : []
+          resolve({ data: { todoList: initialList } })
+        }, 2000)
+      })
+    }
+    loadTodoList().then((result) => {
+      setTodoList(result.data.todoList)
+      setIsLoading(false)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+      console.log(todoList)
+    }
+  }, [todoList, isLoading])
+
+  function addTodo(newTodo) {
+    setTodoList([...todoList, newTodo])
+    console.log(newTodo)
```

Comment:
Please remove `console.log` statements from your code when you're no longer using them for debugging or development.